From: Mohamad Haj Yahia <mohamad@mellanox.com>
Date: Fri, 10 Jun 2016 00:07:38 +0300
Subject: net/mlx5: E-Switch, always set mc_promisc for allmulti vports
Patch-mainline: v4.7-rc3
Git-commit: 62e3c24ac4f0ee307c41a3652636f88a3f8d165c
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

Set the mc_promisc flag also in the case of adding new mc address to
existing allmulti vport.

Fixes: a35f71f27a61 ('net/mlx5: E-Switch, Implement promiscuous rx modes vf request handling')
Signed-off-by: Mohamad Haj Yahia <mohamad@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/eswitch.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/mellanox/mlx5/core/eswitch.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/eswitch.c
@@ -651,6 +651,7 @@ static void update_allmulti_vports(struc
 					esw_fdb_set_vport_rule(esw,
 							       mac,
 							       vport_idx);
+			iter_vaddr->mc_promisc = true;
 			break;
 		case MLX5_ACTION_DEL:
 			if (!iter_vaddr)
